WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -110,6 +114,25 @@ final class AddEditCouponViewModel: ObservableObject {
         codeField = existingCoupon.code
         descriptionField = existingCoupon.description
         expiryDateField = existingCoupon.dateExpires
+        freeShipping = existingCoupon.freeShipping
+        couponRestrictionsViewModel = CouponRestrictionsViewModel(coupon: existingCoupon)
+    }
+
+    /// The method will generate a code in the same way as the existing admin website code does.
+    /// https://github.com/woocommerce/woocommerce/blob/23710744c01ded649d6a94a4eaea8745e543159f/assets/js/admin/meta-boxes-coupon.js#L53
+    /// We will loop to select 8 characters from the set `ABCDEFGHJKMNPQRSTUVWXYZ23456789` at random using `arc4random_uniform` for randomness.
+    /// https://github.com/woocommerce/woocommerce/blob/2e60d47a019a6e35f066f3ef43a56c0e761fc8e3/includes/admin/class-wc-admin-assets.php#L295
+    ///
+    func generateRandomCouponCode() {
+        let dictionary: [String] = "ABCDEFGHJKMNPQRSTUVWXYZ23456789".map { String($0) }
```

Comment:
I just realized we don't include "0", "1", "I", "L", "O" in the character list probably because they are easily mistaken for one another 😄